### ISSUE TITLE
Fix regression for single line comments in Less

### ISSUE THREAD

**Issue by EvHaus** [stylelint/stylelint#3754]
When using the "declaration-block-semicolon-newline-after" rule with "always-multi-line" and the "less" syntax, the rule wants a new line after single-line comments inside any nested block, which causes `stylelint-disable-next-line` comments to become broken because it will request a new line after the comment which would invalidate that disable.

> Which rule, if any, is the bug related to?

`declaration-block-semicolon-newline-after`

> What CSS is needed to reproduce the bug?

```css
table {
	width: 100%;

	// stylelint-disable-next-line selector-max-combinators, selector-max-type
	td {
		padding: 8px;
	}
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-semicolon-newline-after": "always-multi-line"
  }
}
```

> Which version of stylelint are you using?

`9.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint test.less --syntax less`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes. Affected nested rules in LESS

> What did you expect to happen?

No error should be triggered using the example code above.

> What actually happened (e.g. what warnings or errors did you get)?

The following error was flagged:

```shell
test.less
2:14  ✖  Expected newline after ";" in a multi-line declaration block   declaration-block-semicolon-newline-after
```

**Comment by EvHaus** [comment]
This seems to be caused by the `postcss-less@3.x.x` upgrade that occurred in https://github.com/stylelint/stylelint/pull/3687

A slew of other LESS issues are being thrown in our application now as well. Including `at-rule-no-unknown` thinking LESS variables are "at rules".

```
 1:1  ✖  Unexpected unknown at-rule "@arrow-height"   at-rule-no-unknown
```

**Comment by jeddy3** [comment]
@EvHaus Thanks for the report and for using the template.

This looks to be a regression. It could be with the Less parser itself or with how the whitespace rules make use of it. @jwilsson Friendly ping, in case anything springs to mind.

> Including at-rule-no-unknown thinking LESS variables are "at rules".

This was reported in https://github.com/stylelint/stylelint/issues/3749. Unlike your first issue, I think this can definitely be fixed within stylelint itself. Feel free to contribute a fix for this as I believe it's labeled as "help wanted".

**Comment by jeddy3** [comment]
I believe this issue might require a bit of digging. I think the problem might be in how either the [`nextNonCommentNode`](https://github.com/stylelint/stylelint/blob/master/lib/utils/nextNonCommentNode.js) or [`rawNodeString`](https://github.com/stylelint/stylelint/blob/master/lib/utils/rawNodeString.js) utils are interacting with the Less parser.

I won't have time to dig into this deeper and I'm not sure any of the core team use Less themselves. We rely on contributors to continue Less support, so please pick this up if you use Less and would like to see this fixed.

**Comment by jwilsson** [comment]
Hmm, I can't think of anything right off the bat but I can do some further digging across this and the other Less-related issue later this week.

**Comment by jwilsson** [comment]
This issue stemmed from a regression related to how the `Node.raws` properties was populated in `postcss-less`. A fix was just released there so a fresh install of `stylelint` should solve this issue.
